src/data/polynomial/degree/definitions.lean

```diff
@@ -736,6 +736,37 @@ begin
   refine add_le_add _ _; apply degree_le_nat_degree,
 end
 
+lemma nat_degree_pow_le {p : polynomial R} {n : ℕ} : (p ^ n).nat_degree ≤ n * p.nat_degree :=
+begin
+  induction n with i hi,
+  { simp },
+  { rw [pow_succ, nat.succ_mul, add_comm],
+    apply le_trans nat_degree_mul_le,
+    exact add_le_add_left hi _ }
+end
+
+@[simp] lemma coeff_pow_mul_nat_degree (p : polynomial R) (n : ℕ) :
+  (p ^ n).coeff (n * p.nat_degree) = p.leading_coeff ^ n :=
+begin
+  induction n with i hi,
+  { simp },
+  { by_cases hp1 : p.leading_coeff ^ i = 0; repeat { rw pow_succ' },
+    { by_cases hp2 : p ^ i = 0; rw [hp1, zero_mul],
+      { rw [hp2, zero_mul, coeff_zero] },
+      { have h1 : (p ^ i).nat_degree < i * p.nat_degree,
+        { by_contra h,
+          replace h := eq_iff_le_not_lt.mpr ⟨nat_degree_pow_le, h⟩,
+          rw [←h, hp1] at hi,
+          exact (leading_coeff_ne_zero.mpr hp2) hi },
+        have h2 : (p ^ i * p).nat_degree < i.succ * p.nat_degree,
+        by calc (p ^ i * p).nat_degree ≤ (p ^ i).nat_degree + p.nat_degree : nat_degree_mul_le
+                                   ... < i * p.nat_degree + p.nat_degree : nat.add_lt_add_right h1 _
+                                   ... = i.succ * p.nat_degree : by rw ←nat.succ_mul,
+        rw coeff_eq_zero_of_nat_degree_lt h2 } },
+    { rw [nat.succ_mul, ←nat_degree_pow' hp1, ←leading_coeff_pow' hp1],
+      exact coeff_mul_degree_add_degree _ _ } }
```

Comment:
tidied a little; instead of `by_cases h : p; rw [foo, bar]` it's clearer to use `rw [foo, bar], by_cases h : p` as the lean goal view does not handle `;` well. It's also faster.
```suggestion
  induction n with i hi,
  { simp },
  { rw [pow_succ', pow_succ', nat.succ_mul],
    by_cases hp1 : p.leading_coeff ^ i = 0,
    { rw [hp1, zero_mul],
      by_cases hp2 : p ^ i = 0,
      { rw [hp2, zero_mul, coeff_zero] },
      { apply coeff_eq_zero_of_nat_degree_lt,
        have h1 : (p ^ i).nat_degree < i * p.nat_degree,
        { by_contra h,
          replace h := eq_iff_le_not_lt.mpr ⟨nat_degree_pow_le, h⟩,
          rw [←h, hp1] at hi,
          exact (leading_coeff_ne_zero.mpr hp2) hi },
        calc (p ^ i * p).nat_degree ≤ (p ^ i).nat_degree + p.nat_degree : nat_degree_mul_le
                                ... < i * p.nat_degree + p.nat_degree : add_lt_add_right h1 _ } },
    { rw [←nat_degree_pow' hp1, ←leading_coeff_pow' hp1],
      exact coeff_mul_degree_add_degree _ _ } }
```

Reply:
The `have` golfs slightly to
```
        have h1 : (p ^ i).nat_degree < i * p.nat_degree,
        { apply lt_of_le_of_ne nat_degree_pow_le (λ h, hp2 _),
          rw ←h at hi,
          exact leading_coeff_eq_zero.mp hi },
          ```